ngraph/core/reference/include/ngraph/runtime/reference/convert_color_nv12.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+
+#include "openvino/op/util/convert_color_nv12_base.hpp"
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+template <typename T>
+void color_convert_nv12(const T* arg_y,
+                        const T* arg_uv,
+                        T* out_ptr,
+                        size_t batch_size,
+                        size_t image_h,
+                        size_t image_w,
+                        size_t stride_y,
+                        size_t stride_uv,
+                        ov::op::util::ConvertColorNV12Base::ColorConversion color_format) {
+    // With C++20 - std::endian can be used at compile time
+    auto little_endian = []() -> int {
+        union {
+            int32_t i;
+            char c[4];
+        } u = {0x00000001};
+        return static_cast<int>(u.c[0]);
+    };
```

Comment:
why do we need it?

Reply:
As per https://docs.microsoft.com/en-us/windows-hardware/drivers/display/4-2-0-video-pixel-formats
> If these samples are addressed as a little-endian WORD type, Cb would be in the least significant bits and Cr would be in the most significant bits with the same total stride as the Y samples
It is possible to determine endianness at compile time, however see,s like not all platforms have macro like __BYTE_ORDER__, __ORDER_LITTLE_ENDIAN__, etc. So I decided to check this at runtime, overhead looks minimal comparing to complexity of other operations